libraries/AP_Baro/AP_Baro_atmosphere.cpp

```diff
@@ -0,0 +1,345 @@
+    // const float temp =  get_temperature();
+
+    if (is_zero(temp_slope)) {    // Iso-thermal layer
+        const float fac = expf(-GRAVITY_MSS / (temperature_K * R_specific) * (alt_amsl - atmospheric_1976_consts[idx].amsl_m));
+        pressure = atmospheric_1976_consts[idx].pressure_Pa * fac;
+    } else {            // Gradient temperature layer
+        const float fac =  GRAVITY_MSS / (temp_slope * R_specific);
+        const float temp_ratio = temperature_K / atmospheric_1976_consts[idx].temp_K; // temperature ratio [unitless]
+        pressure = atmospheric_1976_consts[idx].pressure_Pa * powf(temp_ratio, -fac);
+    }
+}
+
+/*
+  return air density (kg/m^3), given geometric altitude (m)
+*/
+float AP_Baro::get_air_density_for_alt_amsl(float alt_amsl)
+{
+    alt_amsl = geometric_alt_to_geopotential(alt_amsl);
+
+    const uint8_t idx = find_atmosphere_layer_by_altitude(alt_amsl);
+    const float temp_slope = atmospheric_1976_consts[idx].temp_lapse;
+    const float temp =  get_temperature_by_altitude_layer(alt_amsl, idx);
+    // const float temp =  get_temperature();
+
+    float rho;
+    if (is_zero(temp_slope)) {    // Iso-thermal layer
+        const float fac = expf(-GRAVITY_MSS / (temp * R_specific) * (alt_amsl - atmospheric_1976_consts[idx].amsl_m));
+        rho      = atmospheric_1976_consts[idx].density     * fac;
+    } else {            // Gradient temperature layer
+        const float fac =  GRAVITY_MSS / (temp_slope * R_specific);
+        const float temp_ratio = temp / atmospheric_1976_consts[idx].temp_K; // temperature ratio [unitless]
+        rho      = atmospheric_1976_consts[idx].density     * powf(temp_ratio, -(fac + 1));
+    }
+    return rho;
+}
+
+/*
+  return current scale factor that converts from equivalent to true airspeed
+*/
+float AP_Baro::get_EAS2TAS_extended(float altitude) const
+{
+    float density = get_air_density_for_alt_amsl(altitude);
+    if (!is_positive(density)) {
+        // above this height we are getting closer to spacecraft territory...
+        const uint8_t table_size = ARRAY_SIZE(atmospheric_1976_consts);
+        density = atmospheric_1976_consts[table_size-1].density;
+    }
+    return sqrtf(SSL_AIR_DENSITY / density);
+}
+
+/*
+  Given the geometric altitude (m)
+  return scale factor that converts from equivalent to true airspeed
+  used by SITL only
+*/
+float AP_Baro::get_EAS2TAS_for_alt_amsl(float alt_amsl)
+{
+    const float density = get_air_density_for_alt_amsl(alt_amsl);
+    return sqrtf(SSL_AIR_DENSITY / MAX(0.00001,density));
+}
+
+#endif // HAL_BARO_ATMOSPHERIC_EXTENDED || CONFIG_HAL_BOARD == HAL_BOARD_SITL
+
+/*
+  return geometric altitude difference in meters between current pressure and a
+  given base_pressure in Pascal.
+*/
+float AP_Baro::get_altitude_difference(float base_pressure, float pressure) const
+{
+#if HAL_BARO_ATMOSPHERIC_EXTENDED
```

Comment:
Shouldn't be in `HAL`.

`AP_BARO_1976_STANDARD_ATMOSPHERE_ENABLED`

Reply:
done